packages/components/src/IDReader/readers/QRReader/Scanner.tsx

```diff
@@ -37,34 +40,46 @@ const Scanner = (props: ScannerProps) => {
   const scanner = useRef<QRScanner>()
   const videoElement = useRef<HTMLVideoElement>(null)
   const [qrOn, setQrOn] = useState(true)
-  const { onError, onScan } = props
+  const { onError, onScan, validator } = props
+  const onScanSuccess = useCallback(
```

Comment:
@tahmidrahman-dsi do you think we need similar validator functionality in the transformHttpFieldIntoRequest function for the ESignet http field?

Reply:
I think we can. We can provide a validator function to the http field which will be executed in `transformHttpFieldIntoRequest` that can validate the response data. However we might need a design for that error case

Reply:
> if they have validated themsleves with MOSIP and we are now retrieving their data.
Then we can assume the id is valid.
I guess, we might not need validation for e-signet, then